Individual.h

```diff
@@ -0,0 +1,20 @@
+#ifndef COMMUNITY_DETECTION_AI_INDIVIDUAL_H
+#define COMMUNITY_DETECTION_AI_INDIVIDUAL_H
+
+#include <vector>
+#include <cstdlib>
+#include "Graph.h"
+
+class Individual {
+private:
+    std::vector<int> communities_division_;
```

Comment:
The name communities division implies that you've also created the communities. I believe that something like chosen_neighbour would be better since chosen_neighbour[x] is the chosen neighbour of x 😅